replication_timeout_test.go

```diff
@@ -285,3 +290,147 @@ func TestReplicationRequestIncompleteResponses(t *testing.T) {
 	laggingNode.e.AdvanceTime(laggingNode.e.StartTime.Add(simplex.DefaultReplicationRequestTimeout * 2))
 	laggingNode.storage.waitForBlockCommit(startSeq)
 }
+
+type collectNotarizationComm struct {
+	lock          *sync.Mutex
+	notarizations map[uint64]*simplex.Notarization
+	testNetworkCommunication
+
+	replicationResponses chan struct{}
+}
+
+func newCollectNotarizationComm(nodeID simplex.NodeID, net *inMemNetwork, notarizations map[uint64]*simplex.Notarization, lock *sync.Mutex) *collectNotarizationComm {
```

Comment:
Can we just move all the logic in these ad-hoc structs to a closure instantiated inside the test? 

It feels wrong to have so much logic defined just for one test.